mindsdb/interfaces/database/integrations.py

```diff
@@ -94,19 +94,31 @@ def set(self, handler: DatabaseHandler):
                 pass
             self._start_clean()
 
-    def get(self, name: str) -> Optional[DatabaseHandler]:
+    def get(self, name: str, thread_safe: bool = False) -> Optional[DatabaseHandler]:
```

Comment:
I don't really like to add additional arguments to functions. Better keep interface simple. What if add 'thread_safe' arg to the handler class itself?
Therefore 'set' may look like:
```python
def set(self, handler: DatabaseHandler):
    ...
    key = (handler.name, ctx.company_id, 0 if getattr(handler, 'thread_safe', False) else thread_id)
    ...
```
and 'get' like:
```python
def get(self, name: str) -> Optional[DatabaseHandler]:
    ...
    key = (name, ctx.company_id, thread_id)
    if key not in cache:
        key = (name, ctx.company_id, 0)
    if (
        key not in self.handlers
        or self.handlers[key]['expired_at'] < time()
    ):
        return None
    ...
```
what do you think?

Reply:
@StpMax This makes sense. I will push the suggested changes shortly.